candidate/models.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import unicode_literals
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+from ballot_measure.models import BallotMeasureChoice
+
+
+class SocialMediaModel(models.Model):
+    class Meta:
+        abstract = True
+
+    photo_url = models.ImageField(blank=True)
+    website_url = models.URLField(
+        help_text='URL for the official website.', blank=True)
+    facebook_url = models.URLField(
+        help_text='URL for the Facebook page.', blank=True)
+    twitter_url = models.URLField(
+        help_text='URL for the Twitter page.', blank=True)
+
+
+@python_2_unicode_compatible
+class Party(SocialMediaModel):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The party name.')
+
+    def __str__(self):
+        return "The %s party" % self.name
+
+
+@python_2_unicode_compatible
+class Person(SocialMediaModel):
+    """
+    """
+    first_name = models.CharField(
+        max_length=255, help_text='The candidate\'s first name.')
+    middle_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s middle name.')
+    last_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s last name.')
+    party = models.ForeignKey('Party', blank=True, null=True)
+
+    def __str__(self):
+        return "%s, %s%s" % (
+            self.last_name, self.first_name,
+            '' if self.middle_name is None else ' %s.' % self.middle_name[0])
+
+
+@python_2_unicode_compatible
+class Office(models.Model):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The office name.')
+    description = models.CharField(
+        max_length=1024, help_text='The office description.')
+    locality = models.ForeignKey('locality.Locality')
+
+    def __str__(self):
+        return "Office for %s in %s" % (
+            self.name, self.locality.dereference())
+
+
+@python_2_unicode_compatible
+class Election(models.Model):
```

Comment:
> As for Office vs. Election (as I'm using it here), I would prefer to keep them separate. Office exists independent of election or not. It's like a generic government object.

That makes sense for Office. I'm fine to have an Office entity representing something outside of the election. However, if Election means "an office up for election on a ballot", what term would you propose for the election day event?

The Voter Information Project is writing a [spec](http://vip-specification.readthedocs.org/en/vip5/xml/index.html#elements) to standardize ballot information across the country. IMHO, their terminology avoids a lot ambiguity which is why I would prefer to standardize on their terms (not necessarily their spec). At the very least, I found reading through it helpful to think about the domain space more. Like how they associate Elections with different authorities (city, county, state, federal), which made sense to me which sounded relevant with what we're doing.

VIP is also representing [additional data](http://vip-specification.readthedocs.org/en/vip5/xml/elements/election.html) for their `Election` entity, like election type (city, state, federal) which I think will be relevant to how the contributions are reported (speculating).